internal/pkg/deps/vendor.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package deps
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	gg "github.com/hashicorp/go-getter"
+	"github.com/hashicorp/hcl/v2/hclsimple"
+
+	"github.com/hashicorp/nomad-pack/sdk/pack"
+	"github.com/hashicorp/nomad-pack/terminal"
+)
+
+// Vendor reads the metadata.hcl from the provided directory and downloads
+// dependencies
+func Vendor(ctx context.Context, ui terminal.UI, targetPath string) error {
+	// attempt to read metadata.hcl
+	metadata := &pack.Metadata{}
+	err := hclsimple.DecodeFile(path.Join(targetPath, "metadata.hcl"), nil, metadata)
+	if err != nil {
+		return err
+	}
+
+	if len(metadata.Dependencies) == 0 {
+		return fmt.Errorf("metadata.hcl file does not contain any dependencies")
+	}
```

Comment:
If someone runs the command on a pack with zero dependencies this will return an error? Did we consider this being a noop, or just an info log message?

Reply:
I would consider it an error. The command is specifically intended to download dependencies, i.e., my understanding is that if you run it as a user, you expect the `metadata.hcl` to contain dependency information. If it doesn't contain it, we should return an error.